spec/router_spec.lua

```diff
@@ -41,12 +41,6 @@ describe("Router", function()
         assert.same(dummy.params, {id="2"})
       end)
 
-      it("supports an extension on a param", function()
```

Comment:
I don't see a test testing support for the `.`. I just see removing support for partial matching.

The tests are testing that `/:id.json` extracts `1`. I don't see how that could conflict with `/:email` matching the whole string. It might be not easy because of how it is implemented, but that does not mean it is not possible.

I can't merge such breaking functionality, sorry. I'd be fine merging it, if it would not break current tests and remove existing functionality.